node/pkg/node/node.go

```diff
@@ -0,0 +1,575 @@
+	// Injected VAAs (manually generated rather than created via observation)
+	injectC channelPair[*vaa.VAA]
+	// TODO describe
+	acctC channelPair[*common.MessagePublication]
+}
+
+func NewGuardianNode(
+	env common.Environment,
+	db *db.Database,
+	gk *ecdsa.PrivateKey,
+	wormchainConn *wormconn.ClientConn, // TODO does this need to be here?
+) *G {
+	g := G{
+		env:           env,
+		db:            db,
+		gk:            gk,
+		wormchainConn: wormchainConn,
+	}
+	return &g
+}
+
+// initializeBasic sets up everything that every GuardianNode needs before any options can be applied.
+func (g *G) initializeBasic(logger *zap.Logger, rootCtxCancel context.CancelFunc) {
+	g.rootCtxCancel = rootCtxCancel
+
+	// Setup various channels...
+	g.gossipSendC = make(chan []byte)
+	g.obsvC = make(chan *gossipv1.SignedObservation, inboundObservationBufferSize)
+	g.msgC = makeChannelPair[*common.MessagePublication](0)
+	g.setC = makeChannelPair[*common.GuardianSet](1) // This needs to be a buffered channel because of a circular dependency between processor and accountant during startup.
+	g.signedInC = makeChannelPair[*gossipv1.SignedVAAWithQuorum](inboundSignedVaaBufferSize)
+	g.obsvReqC = makeChannelPair[*gossipv1.ObservationRequest](observationRequestOutboundBufferSize)
+	g.obsvReqSendC = makeChannelPair[*gossipv1.ObservationRequest](observationRequestInboundBufferSize)
+	g.injectC = makeChannelPair[*vaa.VAA](0)
+	g.acctC = makeChannelPair[*common.MessagePublication](accountant.MsgChannelCapacity)
+
+	// Guardian set state managed by processor
+	g.gst = common.NewGuardianSetState(nil)
+
+	// provides methods for reporting progress toward message attestation, and channels for receiving attestation lifecyclye events.
+	g.attestationEvents = reporter.EventListener(logger)
+
+	// allocate maps
+	g.runnablesWithScissors = make(map[string]supervisor.Runnable)
+	g.runnables = make(map[string]supervisor.Runnable)
+}
+
+// applyOptions applies `options` to the GuardianNode.
+// Each option must have a unique option.name.
+// If an option has `dependencies`, they must be defined before that option.
+func (g *G) applyOptions(ctx context.Context, logger *zap.Logger, options []*GuardianOption) error {
+	configuredComponents := make(map[string]struct{}) // using `map[string]struct{}` to implement a set here
+
+	for _, option := range options {
+		// check that this component has not been configured yet
+		if _, ok := configuredComponents[option.name]; ok {
+			return fmt.Errorf("Component %s has been configured already and cannot be configured a second time.", option.name)
+		}
+
+		// check that all dependencies have been met
+		for _, dep := range option.dependencies {
+			if _, ok := configuredComponents[dep]; !ok {
+				return fmt.Errorf("Component %s requires %s to be configured first. Check the order of your options.", option.name, dep)
+			}
+		}
+
+		// run the config
+		err := option.f(ctx, logger, g)
+		if err != nil {
+			return err
```

Comment:
Can we include the component with the error just in case the lower layer isn't including the component name?

Reply:
done